src/main/scala/abalone/format/FEN.scala

```diff
@@ -1,33 +1,60 @@
 package strategygames.abalone.format
 
 import strategygames.Player
+import strategygames.abalone.File
+import strategygames.abalone.Piece
 import strategygames.abalone.PieceMap
+import strategygames.abalone.Pos
+import strategygames.abalone.Stone
 
 final case class FEN(value: String) extends AnyVal {
+  // squares are described from topLeft to bottomRight in the FEN :
+  // eg. Belgian Daisy: SS1ss/SSSsss/1SS1ss1/8/9/8/1ss1SS1/sssSSS/ss1SS
+  // Snakes: sssss/s5/s6/s1SSSSS1/1s5S1/1sssss1S/6S/5S/SSSSS
 
   override def toString = value
 
-  def player: Option[Player] =
-    value.split(' ').lift(3) flatMap (_.headOption) flatMap Player.apply
+  def pieces: PieceMap = squares.zipWithIndex.map {
+      case (pieces, row) => pieces.zipWithIndex.map[Option[(Option[strategygames.abalone.Pos], strategygames.abalone.Piece)]] { 
+        case (piece, index) => piece match {
+          case Stone.forsyth => Some((Pos.at(index, (File.all.size - 1 - row)), new Piece(strategygames.abalone.P1, Stone)))
```

Comment:
I was hoping for pos.at(file, rank), or pos.at(index) 
using terminology from both is confusing. I think its just needs a rename. index -> file?

Reply:
thanks indeed it will be better